service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/ResourceExtractor.java

```diff
@@ -0,0 +1,18 @@
+package uk.nhs.adaptors.gp2gp.ehr.utils;
+
+import java.util.Optional;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Resource;
+
+public class ResourceExtractor {
+
+    public static Optional<Resource> extractResourceByReference(Bundle allBundle, String reference) {
+        return allBundle.getEntry()
```

Comment:
Similar part to https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/82 , this method could be used by EncounterExtractor depends which one will go first.